File: packages/jsii-pacmak/test/expected.java/java/org/jsii/tests/calculator/NumberGenerator.java
package org.jsii.tests.calculator;
/**
 * This allows us to test that a reference can be stored for objects that
 * implement interfaces.
 */
@org.jsii.Jsii(module = org.jsii.tests.calculator.$Module.class, fqn = "jsii$jsii_calc$.NumberGenerator")
public class NumberGenerator extends org.jsii.JsiiObject {
    protected NumberGenerator(final org.jsii.JsiiObject.InitializationMode mode) {
        super(mode);
    }
    public NumberGenerator(final org.jsii.tests.calculator.IRandomNumberGenerator generator) {
        super(org.jsii.JsiiObject.InitializationMode.Jsii);
        org.jsii.JsiiEngine.getInstance().createNewObject(this, java.util.stream.Stream.of(generator).toArray());
    }
    public java.lang.Number nextTimes100() {
        return this.jsiiCall("nextTimes100", java.lang.Number.class);
    }
    public java.lang.Boolean isSameGenerator(final org.jsii.tests.calculator.IRandomNumberGenerator gen) {
        return this.jsiiCall("isSameGenerator", java.lang.Boolean.class, java.util.stream.Stream.of(gen).toArray());
    }
    public org.jsii.tests.calculator.IRandomNumberGenerator getGenerator() {
        return this.jsiiGet("generator", org.jsii.tests.calculator.IRandomNumberGenerator.class);
    }
    public void setGenerator(final org.jsii.tests.calculator.IRandomNumberGenerator value) {
        this.jsiiSet("generator", value);
    }
}

File: packages/jsii-dotnet-runtime/test/AWS.Jsii.Tests.Calculator/NumberGenerator.cs
using AWS.Jsii.Runtime.Deputy;

namespace AWS.Jsii.Tests.Calculator
{
    /// <summary>
    /// This allows us to test that a reference can be stored for objects that
    /// implement interfaces.
    /// </summary>
    [JsiiClass(typeof(NumberGenerator), "jsii-calc.NumberGenerator", "[{\"name\":\"generator\",\"type\":{\"fqn\":\"jsii-calc.IRandomNumberGenerator\"}}]")]
    public class NumberGenerator : DeputyBase
    {
        public NumberGenerator(IIRandomNumberGenerator generator): base(new DeputyProps(new object[]{generator}))
        {
        }

        protected NumberGenerator(ByRefValue reference): base(reference)
        {
        }

        protected NumberGenerator(DeputyProps props): base(props)
        {
        }

        [JsiiProperty("generator", "{\"fqn\":\"jsii-calc.IRandomNumberGenerator\"}")]
        public virtual IIRandomNumberGenerator Generator
        {
            get => GetProperty<IIRandomNumberGenerator>();
            set => SetProperty(value);
        }

        [JsiiMethod("nextTimes100", "{\"primitive\":\"number\"}", "[]")]
        public virtual double NextTimes100()
        {
            return InvokeMethod<double>(new object[]{});
        }

        [JsiiMethod("isSameGenerator", "{\"primitive\":\"boolean\"}", "[{\"name\":\"gen\",\"type\":{\"fqn\":\"jsii-calc.IRandomNumberGenerator\"}}]")]
        public virtual bool IsSameGenerator(IIRandomNumberGenerator gen)
        {
            return InvokeMethod<bool>(new object[]{gen});
        }
    }
}